velox/serializers/PrestoSerializer.cpp

```diff
@@ -4220,12 +4220,19 @@ void PrestoVectorSerde::deserialize(
     auto compressBuf = folly::IOBuf::create(header.compressedSize);
     source->readBytes(compressBuf->writableData(), header.compressedSize);
     compressBuf->append(header.compressedSize);
+
+    // Process chained uncompressed results IOBufs.
     auto uncompress =
         codec->uncompress(compressBuf.get(), header.uncompressedSize);
-    ByteRange byteRange{
-        uncompress->writableData(), (int32_t)uncompress->length(), 0};
+    std::vector<ByteRange> byteRanges;
```

Comment:
Not sure if we have similar utility for this. Can you add a utility in velox/common/base/IOUtil.h for this with unit test? Like 
```
std::vector<ByteRange> = writableByteRangesFromIOBuf?
```